src/crafting.cpp

```diff
@@ -2277,8 +2277,10 @@ void crafting::complete_disassemble( Character &who, const iuse_location &target
 
     if( !dis.learn_by_disassembly.empty() && !who.knows_recipe( &dis ) ) {
         if( who.can_learn_by_disassembly( dis ) ) {
-            // TODO: make this depend on intelligence
-            if( one_in( 4 ) ) {
+            const SkillLevelMap &char_skills = who.get_all_skills();
+            float skill_bonus = ( 1.0f + char_skills.exceeds_recipe_requirements( dis ) ) * std::max( 1,
```

Comment:
```suggestion
            float skill_bonus = ( 1.0f + char_skills.exceeds_recipe_requirements( dis ) ) * std::max( 1/0f,
```
the error happens because you're comparing between int and float. please cast it.